tools/slicec-cs/src/decoding.rs

```diff
@@ -382,27 +376,25 @@ fn decode_stream_parameter(type_ref: &TypeRef, namespace: &str, encoding: Encodi
     }
 }
 
-fn decode_result_field(type_ref: &TypeRef, namespace: &str, encoding: Encoding) -> CodeBlock {
-    let mut decode_func = match type_ref.is_optional {
-        true => decode_func_body(type_ref, namespace, encoding),
-        false => decode_func(type_ref, namespace, encoding),
+/// Used by `decode_dictionary` and `decode_result` to generate a decoding action that includes a cast for collections.
```

Comment:
I would remove where it's used. Eventually we'll use it somewhere else and never update this comment.

Reply:
I completely rewrote the doc comment.